src/EnergyPlus/SimAirServingZones.cc

```diff
@@ -2941,7 +2947,9 @@ namespace SimAirServingZones {
 
 			ManageControllers( ControllerName, ControllerIndex, FirstHVACIteration, AirLoopNum, iControllerOpIterate, ControllerConvergedFlag, IsUpToDateFlag, BypassOAController );
 
-			PrimaryAirSystem( AirLoopNum ).ControlConverged( ControllerIndex ) = ControllerConvergedFlag;
+			if ( HVACControllers::ControllerProps( ControllerIndex ).AirLoopControllerIndex > 0 ) {
+				PrimaryAirSystem( AirLoopNum ).ControlConverged( HVACControllers::ControllerProps( ControllerIndex ).AirLoopControllerIndex ) = ControllerConvergedFlag;
```

Comment:
This line right here, 2951 in this branch, is where it failed. This is an array of convergence for each air loop. But each air loop has multiple controllers AND can have an OA system with multiple controllers. So what the converged flag needs to be is an aggregate of all controllers on this air loop. I don't think this flag represents what it should. Seems it should be an .AND. of air loop and OS sys controllers.

Reply:
The fix was to save the index to the controller for each air loop. But this is confusing and I'm not sure this works correctly.

Reply:
Also, this can probably be done somewhere just once to speed things up. Not sure yet where that might be.